### PR TITLE
Added jsonsettings support

### PR DESCRIPTION
**Pull request by ellmetha**
* added support for JSON "env settings"
* refactored the whole role

### CODE REVIEW

File: tasks/manage.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+- block:
+  - name: Export Django project's required environment variables
+    shell: "export {{ item.name }}={{ item.value }}"
```
Comment (ghost) [inline tasks/manage.yml]: The goal of this line is to set environment variable for migrate and collectstatic actions below? I'm surprised that it works at all and I'm not sure that ansible guarantees that all tasks from a given role are executed in the same shell session. Shouldn't we use something like http://docs.ansible.com/ansible/playbooks_environment.html instead?

File: tasks/manage.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+- block:
```
Comment (ghost) [inline tasks/manage.yml:1]: This block isn't needed

File: tasks/config_jsonsettings.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+- name: Create .env.json
+  template: src=.env.json.j2 dest={{ django_path }}/.env.json
```
Comment (ghost) [inline tasks/config_jsonsettings.yml]: You don't need to use templating if the contents of your dest file is in a variable. You can use the `content` option of `copy`: http://docs.ansible.com/ansible/copy_module.html

File: defaults/main.yml [hunk 4]
```diff
@@ -42,3 +44,73 @@ django_touch_reload_path: "{{ django_project_path }}"
...
+# Domain names under which the project is served. Ends up in ALLOWED_HOSTS.
+django_domain_name: "{{ django_name }}.vagrant.dev"
+django_extra_domain_names: []
+# Arbitrary python code to add to local_settings.py. Will be appended as-is to the end of the file.
+django_env_settings_extra: ''
+
+# The django_jsonsettings_properties is only used when django_settings_pattern is set to
+# "jsonsettings". It must be a dictionary of key/values that should be injected into the project's
+# settings module using the "get_envsetting('SECRET_KEY')" pattern.
+django_jsonsettings_properties: {
```
Comment (ghost) [inline defaults/main.yml]: Why the special syntax and indent inconsistency? a yaml hash is usually expressed as:

```
foo:
  bar: 42
  baz: 12
```

File: tasks/main.yml [hunk 5]
```diff
@@ -1,5 +1,20 @@
 - block:
-  - include: django.yml
+  - include: virtualenv.yml
+
+  - name: Verify that we have a valid django_settings_pattern
```
Comment (ghost) [inline tasks/main.yml]: Practically the same thing, but `assert` is a bit more fitting for this: http://docs.ansible.com/ansible/assert_module.html

File: tasks/manage.yml (hunk 2 above)
Comment (ellmetha) [reply tasks/manage.yml:1]: Now it is!

File: tasks/config_jsonsettings.yml (hunk 3 above)
Comment (ellmetha) [reply tasks/config_jsonsettings.yml]: Fixed!

File: defaults/main.yml (hunk 4 above)
Comment (ellmetha) [reply defaults/main.yml]: Fixed!

File: tasks/main.yml (hunk 5 above)
Comment (ellmetha) [reply tasks/main.yml]: Fixed!

File: tasks/config_jsonsettings.yml [hunk 6]
```diff
@@ -1,5 +1,7 @@
 - name: Create .env.json
-  template: src=.env.json.j2 dest={{ django_path }}/.env.json
+  copy:
+    content: "{{ django_jsonsettings_properties }}"
```
Comment (ghost) [inline tasks/config_jsonsettings.yml]: `|to_nice_json(indent=2)` is gone?

File: tasks/main.yml [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
   - include: virtualenv.yml
 
-  - name: Verify that we have a valid django_settings_pattern
-    fail:
-      msg: "django_settings_pattern must be either 'localsettings', 'jsonsettings' or 'none'"
-    when: "django_settings_pattern not in ['localsettings', 'jsonsettings', 'none']"
+  - assert:
+      that:
+        - "django_settings_pattern in ['localsettings', 'jsonsettings', 'none']"
+      msg: "'my_param' must be between 0 and 100"
```
Comment (ghost) [inline tasks/main.yml]: wrong msg

File: tasks/config_jsonsettings.yml (hunk 6 above)
Comment (ellmetha) [reply tasks/config_jsonsettings.yml]: `copy` seems to ignore indentation... Not really required, though.

File: tasks/main.yml (hunk 7 above)
Comment (ellmetha) [reply tasks/main.yml]: Fixed :wink:

**Pull request merged**